algorithms/kernel/elastic_net/elastic_net_train_dense_default_impl.i

```diff
@@ -0,0 +1,288 @@
+            result |= daal::services::internal::daal_memcpy_s(yTrainPtr, nDependentVariables * nRows * sizeof(algorithmFPType), yPtr,
+                                                              nDependentVariables * nRows * sizeof(algorithmFPType));
+            DAAL_CHECK(!result, services::ErrorMemoryCopyFailedInternal);
+        }
+
+        daal::internal::WriteRows<algorithmFPType, cpu> xBD(xTrain.get(), 0, nRows);
+        DAAL_CHECK_BLOCK_STATUS(xBD);
+        daal::internal::WriteRows<algorithmFPType, cpu> yBD(yTrain.get(), 0, nRows);
+        DAAL_CHECK_BLOCK_STATUS(yBD);
+        algorithmFPType * xPtr = xBD.get();
+        algorithmFPType * yPtr = yBD.get();
+
+        algorithmFPType inversedNRows = (algorithmFPType)1.0 / (algorithmFPType)nRows;
+        yMeans.reset(nDependentVariables);
+        yMeansPtr = yMeans.get();
+        DAAL_CHECK_MALLOC(yMeansPtr);
+        for (size_t i = 0; i < nDependentVariables; i++) //[TBD] do in parallel
+        {
+            yMeansPtr[i] = 0;
+        }
+        for (size_t i = 0; i < nRows; i++)
+        {
+            PRAGMA_IVDEP
+            PRAGMA_VECTOR_ALWAYS
+            for (size_t id = 0; id < nDependentVariables; ++id)
+            {
+                yMeansPtr[id] += yPtr[i * nDependentVariables + id];
+            }
+        }
+        for (size_t i = 0; i < nDependentVariables; ++i)
+        {
+            yMeansPtr[i] *= inversedNRows;
+        }
+        for (size_t i = 0; i < nRows; i++)
+        {
+            for (size_t id = 0; id < nDependentVariables; ++id)
+            {
+                yPtr[i * nDependentVariables + id] -= yMeansPtr[id];
+            }
+        }
+
+        xMeans.reset(nFeatures);
+        xMeansPtr = xMeans.get();
+        DAAL_CHECK_MALLOC(xMeansPtr);
+
+        for (size_t i = 0; i < nFeatures; ++i) xMeansPtr[i] = 0;
+
+        const size_t blockSize = 256;
+        size_t nBlocks         = nRows / blockSize;
+        nBlocks += (nBlocks * blockSize != nRows);
+
+        DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(size_t, nFeatures, sizeof(algorithmFPType));
+
+        TlsMem<algorithmFPType, cpu, services::internal::ScalableCalloc<algorithmFPType, cpu> > tlsData(nFeatures);
+        daal::threader_for(nBlocks, nBlocks, [&](const size_t iBlock) {
+            algorithmFPType * sum  = tlsData.local();
+            const size_t startRow  = iBlock * blockSize;
+            const size_t finishRow = (iBlock + 1 == nBlocks ? nRows : (iBlock + 1) * blockSize);
+            for (size_t i = startRow; i < finishRow; i++)
+            {
+                PRAGMA_IVDEP
+                PRAGMA_VECTOR_ALWAYS
+                for (int j = 0; j < nFeatures; j++)
+                {
+                    sum[j] += xPtr[i * nFeatures + j];
+                }
+            }
+        });
+        tlsData.reduce([&](algorithmFPType * localSum) {
+            PRAGMA_IVDEP
+            PRAGMA_VECTOR_ALWAYS
+            for (int j = 0; j < nFeatures; j++)
```

Comment:
size_t

Reply:
it was fixed in all cases in the algorithm